programs/omnisol/src/instructions/update_pool.rs

```diff
@@ -0,0 +1,58 @@
+use anchor_lang::prelude::*;
+
+use crate::state::{Pool, Manager};
+use crate::utils::fee::check_fee;
+
+pub fn handle(
+    ctx: Context<UpdatePool>,
+    fee_receiver: Option<Pubkey>,
```

Comment:
Please use separate struct `UpdatePoolData` with all fields.
Something like this `handle(ctx: Context<UpdatePool>, data: UpdatePoolData)`